corehq/motech/repeaters/dbaccessors.py

```diff
@@ -153,6 +154,34 @@ def _get_repeater_ids_by_domain(domain):
     return [result['id'] for result in results]
 
 
+def get_repeat_records_for_ids(doc_ids):
+    from .models import RepeatRecord
+    return [RepeatRecord.wrap(doc['doc']) for doc in get_docs(RepeatRecord.get_db(), doc_ids)]
```

Comment:
I agree. Even if `doc_ids` have a fixed length like 1000, yielding wrapped docs one at a time rather than wrapping them all and storing that in memory in a list will be more memory efficient. (If doc_ids is expected to be very large, which as written here it seems to be, but which I recommend against, then this may be the difference between running smoothly and running out of memory.)

Reply:
I agree too, however I noticed this in the `get_docs` call. If a list comprehension is used underneath, does it matter what is used on top? https://github.com/dimagi/commcare-hq/blob/579b52f1037d30c8e9184214f7278bb1d6a8b9d2/corehq/ex-submodules/dimagi/utils/couch/bulk.py#L123

Reply:
scratch that, it would double the memory demand to use an additional list comprehension on top.